src/Compilers/CSharp/Portable/Symbols/Source/LambdaSymbol.cs

```diff
@@ -375,13 +375,13 @@ public sealed override bool Equals(Symbol symbol, TypeCompareKind compareKind)
         {
             if ((object)this == symbol) return true;
 
-            var lambda = symbol as LambdaSymbol;
-            return (object)lambda != null
+            return symbol is LambdaSymbol lambda
                 && lambda._syntax == _syntax
                 && lambda._refKind == _refKind
                 && TypeSymbol.Equals(lambda.ReturnType, this.ReturnType, compareKind)
-                && System.Linq.ImmutableArrayExtensions.SequenceEqual(lambda.ParameterTypesWithAnnotations, this.ParameterTypesWithAnnotations, (p1, p2) => p1.Type.Equals(p2.Type))
-                && Equals(lambda.ContainingSymbol, this.ContainingSymbol);
+                && ImmutableArrayExtensions.SequenceEqual(ParameterTypesWithAnnotations, lambda.ParameterTypesWithAnnotations, compareKind,
+                                                          (p1, p2, compareKind) => p1.Equals(p2, compareKind))
```

Comment:
Can we use `System.Linq.ImmutableArrayExtensions.SequenceEqual(..., ..., IEqualityComparer<> comparer)` instead, with `SymbolEqualityComparer`?